src/frontend/packages/core/src/core/utils.service.ts

```diff
@@ -17,30 +17,12 @@ export type OptionalKeys<T extends object> = Exclude<{
   : never
 }[keyof T], undefined>
 
-
-export type NonOptionalKeys<T extends object> = Exclude<{
```

Comment:
All of these meta types, for neatness, should probably be kept together. In this world it should be a separate file in store?
- OptionalKeys
- NonOptionalKeys
- NeverKeys
- KnownKeys
- FilteredByReturnType
- FilteredByNotReturnType
- FilteredByValueType